setuptools/tests/test_build_meta.py

```diff
@@ -440,18 +595,28 @@ def run():
     def test_sys_argv_passthrough(self, tmpdir_cwd):
         path.build(self._sys_argv_0_passthrough)
         build_backend = self.get_build_backend()
-        with pytest.raises(AssertionError):
-            build_backend.build_sdist("temp")
```

Comment:
Is is necessary that the regular `build_meta` backend fails if the `setup.py` script relies on `sys.argv[0]`? What is the reasoning behind this condition?

(Or was it something that was observed in practice and simply added to the test, but not required for building packages correctly?)

Reply:
(The reason for this question is that the code is simply relying on `distutils.core.run_setup` which automatically uses `script_name` as `sys.argv[0]`)

Reply:
I don't know the answer to the question. Since there's no comment, I'd suggest to trace the code and review the change and maybe ask the author. Alternatively, if you estimate that investigation is not worth the time, just make the change but be prepared to address the fallout.

Reply:
I asked the original author (@aimileus) in https://github.com/pypa/setuptools/commit/b2701fb39252fc68b4f0b22c3a3b79039f4cc58e#r66325019

Reply:
Because of the problems found with `pybind11` (that uses a context manager around `setup()` invocation), I had to refrain from changing the `build_meta`.

In the current implementation, the integration rely only on the internals of the `Distribution` class.